include/nuttx/fs/fs.h

```diff
@@ -183,6 +185,7 @@ struct statfs;
 struct pollfd;
 struct mtd_dev_s;
 struct tcb_s;
+struct uio;
```

Comment:
should we move `struct uio` definition and `iovec_total_len` prototype to include/nuttx/fs/fs.h and remove include/nuttx/fs/uio.h?

Reply:
uio itself is not necessarily tied to filesystem.
i prefer to keep it separate.
maybe rename to include/nuttx/uio.h.

Reply:
Ok, but it's better to include nuttx/uio.h in fs.h, to avoid all implementation include it again and again.

Reply:
but could you update in this patch? it's noise to see the temp change in history.

Reply:
i will update this PR later, i meant.

Reply:
done. i will squash the commit later. (probably after you review it.)